hack/e2e/ensure_cmctl.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+USR_LOCAL_BIN="/usr/local/bin/"
+OS=$(go env GOOS)
+ARCH=$(go env GOARCH)
+MINIMUM_CMCTL_VERSION=v1.13.0
+
+verify_cmctl_version() {
+  if ! [ -x "$(command -v cmctl)" ]; then
+      if [[ "${OSTYPE}" == "linux-gnu" ]]; then
+          echo "cmctl not found, installing"
+          curl -fsSL -o cmctl.tar.gz "https://github.com/cert-manager/cert-manager/releases/download/${MINIMUM_CMCTL_VERSION}/cmctl-${OS}-${ARCH}.tar.gz"
```

Comment:
I would love if this was tied to a version and pinned with SHA. Let's do that in follow-up.